octodns/provider/yaml.py

```diff
@@ -102,7 +107,97 @@ def _apply(self, plan):
         if not isdir(self.directory):
             makedirs(self.directory)
 
+        self._do_apply(desired, data)
+
+    def _do_apply(self, desired, data):
         filename = join(self.directory, '{}yaml'.format(desired.name))
         self.log.debug('_apply:   writing filename=%s', filename)
         with open(filename, 'w') as fh:
             safe_dump(dict(data), fh)
+
+
+# Any record name added to this set will be included in the catch-all file,
+# instead of a file matching the record name.
+_CATCHALL_RECORD_NAMES = ('*', '')
+
+
+def list_all_yaml_files(directory):
```

Comment:
General practice in octoDNS is to prefix internal helper methods like this with `_`.  Alternatively it'd be fine to make it a `@classmethod` on `SplitYamlProvider`.

Reply:
I opted for the `_`-prefixed approach.